codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -525,6 +525,15 @@ private class ServerHttpProtocolImplGenerator(
                 *codegenScope,
                 "parser" to parser,
             )
+        } else if (streamingMember != null) {
+            // TODO Need contentTypeCheck here?
```

Comment:
Why wouldn't we?

Reply:
@david-perez The previous case uses getContentTypeCheck() to verify the content-type is json/xml which is parseable by the protocol. But with blob streams the content-type can be anything at all... So I think this TODO can just be removed and we won't need to verify the content-type in the streaming case.

Reply:
@guymguym `@streaming` can only be applied to blob or union, and the specs say what `Content-Type` to use in those cases ([example for AWS XML](https://awslabs.github.io/smithy/1.0/spec/aws/aws-restxml-protocol.html#content-type)).

Reply:
That won't work for s3 which relies on the sent content-type header as the object type...

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax

Reply:
@guymguym Ah, crap. I'm guessing you already verified the S3 service is accepting any MIME type for PutObject's `Content-Type`. Do they also return the MIME type you provide in PutObject when calling GetObject, or do they always return `application/octet-stream`? This seems like it's another of these instances where S3 deviates from the protocol they say they adhere to (AWS restXml). S3 is one of the earliest AWS services and as such I suspect you will encounter these idiosyncrasies if you're trying to build a service that mimics its behavior 100%.

My idea to support these deviations would be to add "escape-hatches" in [`ServerCodegenConfig`](https://github.com/awslabs/smithy-rs/blob/main/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerRustSettings.kt#L27) that would slightly modify codegen behavior (e.g. a `doNotCheckContentType` boolean). In the beginning we were using `CodegenConfg` (i.e. the client's codegen settings) but we realized we were going to need a completely different set of settings, so we refactored in https://github.com/awslabs/smithy-rs/pull/995 to start having our own data class. However, we soon realized that `CodegenConfig` is used everywhere in the codebase, which would thus need a large refactor, so our `ServerCodegenConfig` just converts to a `CodegenConfig` under the hood 😅.

I'm OK with not checking `Content-Type` _for the moment_ in this PR, since that refactor is probably large and outside the scope of this PR, but just bear in mind that we have to adhere to the specs, so it eventually will have to go in; I'm working on adding [a suite of protocol tests](https://github.com/awslabs/smithy/tree/main/smithy-aws-protocol-tests/model/restJson1) for restJson1 and we will then have to whittle down the test failures until we are fully compliant.

Reply:
@david-perez Yeah I know, S3 is a bit different, and I also understand your approach with "escape-hatches" and it sounds like a perfect fit for my use case because I have no problems including s3 specific logic in my code, while I am just trying to get all the generic smithy codegen automated. I'll check out ServerCodegenConfig.

And btw thanks for opening the issue for smithy about the s3 model - I wonder what is the answer - if the model itself was not fully translated to smithy, or perhaps it's focused on the client perspective of the service and these are the requirements they want from *new* clients.